Fix typos in README & add Makefile target for api generation (#62)
Azure/ARO-HCP: api/docs/README.md

This README file provides documentation for the HCP RP API outline, the technologies used to build it, and the steps required to get started with the API.

## Overview

The API is designed using [typespec](https://typespec.io/) that is used to generate the swagger definition.
It also utilizes the [Microsoft typespec libraries](https://azure.github.io/typespec-azure/).

The goal is to create the HCP RP API definition in a Microsoft compliant way.

## Setup

When used from within this project with [VSCode remote extensions](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack),
there is no need to setup anything. The whole environment is already bootstrapped in a container.

> The container is based in Go 1.22 to allow development of the RP as well as it has

Reviewer: ```suggestion
The container includes Go 1.22 to allow development of the RP as well as
```

nodejs, which is required for typespec. The container also has typespec with all required libraries installed. The definition can be found in `.devcontainer/postCreate.sh`, where the libraries are pinned to the latest working versions. Please note, when upgrading version of one, the other might need to be upgraded as well because they tend to break.

> If you have fresh container, you need to run `tsp code install` to enable the VSCode typespec extension. This will give you autocomplete and linting.

Reviewer: ```suggestion
If you have a fresh container, you need to run `tsp code install` to enable the VSCode typespec extension. This will give you autocomplete and linting.
```
